cpg-core/src/main/java/de/fraunhofer/aisec/cpg/TranslationConfiguration.java

```diff
@@ -286,8 +291,14 @@ public Builder sourceLocations(List<File> sourceLocations) {
       return this;
     }
 
-    public Builder loadCompilationDatabase(Map<File, List<String>> compilationDatabase) {
-      this.compilationDatabase = compilationDatabase;
+    public Builder useCompilationDatabase(File compilationDatabaseFile) {
+      Map<File, List<String>> compilationDatabaseForTheFile =
+          CompilationDB.Companion.getCompilationDatabaseFromTheFile(compilationDatabaseFile);
+      if (compilationDatabaseForTheFile != null) {
+        this.compilationDatabase = compilationDatabaseForTheFile;
```

Comment:
This should be done in the CXX language frontend. The builder should only store the file name of the compilation database in the config

Reply:
@oxisto  This is done so that the list of files that need to be analysed by the cpg can be obtained.

Reply:
I know, but the configuration builder should be without any side-effects. It should purely store the necessary information in the configuration (the file to the compilation data base). The language frontend or the translation manager should then use this information to read the compilation database (which has side effects) and set the necessary paths for parsing the files.